src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionUsagesValue.java

```diff
@@ -46,19 +56,63 @@ public abstract class SingleExtensionUsagesValue implements SkyValue {
   /** The repo mappings to use for each module that used this extension. */
   public abstract ImmutableMap<ModuleKey, RepositoryMapping> getRepoMappings();
 
-  public static SingleExtensionUsagesValue create(
-      ImmutableMap<ModuleKey, ModuleExtensionUsage> extensionUsages,
-      String extensionUniqueName,
-      ImmutableList<AbridgedModule> abridgedModules,
-      ImmutableMap<ModuleKey, RepositoryMapping> repoMappings) {
-    return new AutoValue_SingleExtensionUsagesValue(
-        extensionUsages, extensionUniqueName, abridgedModules, repoMappings);
+  abstract Builder toBuilder();
+
+  public static Builder builder() {
+    return new AutoValue_SingleExtensionUsagesValue.Builder();
+  }
+
+  /**
+   * Turns the given usages value for a particular extension into a hash that can be compared for
+   * equality with another hash obtained in this way and compares equal only if the two values are
+   * equivalent for the purpose of evaluating the extension.
+   */
+  static byte[] hashForEvaluation(Gson gson, SingleExtensionUsagesValue usagesValue) {
+    return Hashing.sha256()
+        .hashUnencodedChars(gson.toJson(usagesValue.trimForEvaluation()))
+        .asBytes();
+  }
+
+  /**
+   * Returns a new value with all information removed that does not influence the evaluation of the
+   * extension.
+   */
+  SingleExtensionUsagesValue trimForEvaluation() {
+    return toBuilder()
+        .setExtensionUsages(
+            ImmutableMap.copyOf(
+                Maps.transformValues(
+                    getExtensionUsages(), ModuleExtensionUsage::trimForEvaluation)))
+        // The unique name of the extension is not accessible to the extension's implementation
```

Comment:
could we maybe explicitly hash the usages and abridged modules, instead of "trimming" things from SEUV? Reason I'm asking is that stuff like the comment on line 37 is very easily missable. I think in this case we'd more likely include more fields unintentionally than miss necessary fields.